src/main/java/com/dataartisans/flink/dataflow/translation/wrappers/streaming/io/UnboundedSourceWrapper.java

```diff
@@ -87,6 +86,7 @@ public void run(SourceContext<WindowedValue<T>> ctx) throws Exception {
 				context.collectWithTimestamp(makeWindowedValue(item, timestamp), timestamp.getMillis());
 			}
 
+			// TODO: This will run the loop only until the underlying reader first indicates input has stalled.
 			// try to go to the next record
```

Comment:
That makes sense. The JavaDoc indicates `false` for "no more input available" but it also states that you can try again forever. We should change that then.